feat: add version bumping for v1 recipes

wolfv opened the pull request:
Another version of version bumping for v1 recipes. This time it should handle all the cases :)

Needs: https://github.com/prefix-dev/rattler-build-conda-compat/pull/57

With the pixi toml attached below, you can try this with:

```
mkdir trash
cd trash && git clone https://github.com/conda-forge/jolt-physics-feedstock && cd ../
pixi run bump ./trash/jolt-physics-feedstock/ 5.2.0
cd  trash/jolt-physics-feedstock && git diff
```

Should show you an updated recipe with version & hash bumped to 5.2.0

I am using this `pixi.toml` locally: 

<details>
  <summary>pixi.toml</summary>

```toml
[project]
authors = ["Wolf Vollprecht <w.vollprecht@gmail.com>"]
channels = ["conda-forge"]
description = "Add a short description here"
name = "cf-scripts"
platforms = ["osx-arm64"]
version = "0.1.0"

[tasks]
pip = { cmd = "pip install -e ../../rattler-build-conda-compat --no-deps", outputs = [".pixi.toml"] }
bump = { cmd = "python ./conda_forge_tick/update_recipe/version.py", depends-on = ["pip"] }

[dependencies]
attrs = "*"
backoff = "*"
black = "*"
cachecontrol = "*"
cachetools = "*"
click = "*"
conda = "*"
conda-forge-feedstock-check-solvable = ">=0.4.0"
conda-forge-pinning = "*"
conda-libmamba-solver = "*"
conda-forge-metadata = ">=0.3.0"
conda-package-handling = "*"
conda-smithy = "*"
conda-forge-feedstock-ops = ">=0.9.0"
conda-build = ">=3.24.0"
curl = "*"
depfinder = "*"
distributed = "*"
doctr = "*"
feedparser = "*"
frozendict = "*"
git = "*"
"github3.py" = "*"
grayskull = ">=2.5"
jinja2 = "*"
lockfile = "*"
mamba = ">=0.23"
msgpack-python = "*"
networkx = "!=2.8.1"
numpy = "*"
packaging = "*"
psutil = "*"
pygithub = "*"
pymongo = "*"
pynamodb = "*"
python-dateutil = "*"
python-graphviz = "*"
python-rapidjson = "*"
requests = "*"
rever = "*"
"ruamel.yaml" = "*"
"ruamel.yaml.jinja2" = "*"
scipy = "*"
setuptools = "*"
setuptools_scm = ">=7"
stopit = "*"
streamz = "*"
tar = "*"
tqdm = "*"
wurlitzer = "*"
yaml = "*"
pandas = "*"
xonsh = "*"

[pypi-dependencies]
conda-forge-tick = { path = ".", editable = true }
rattler-build-conda-compat = { path = "../../rattler-build-conda-compat", editable = true }
```
</details>

wolfv commented:
I just added a first test that demonstrates that talking to the pypi API also works for the new recipe format.

I'll start cutting a release over on rattler-build-conda-compat and then we can move forward here. We can port some more tests over.

minrk commented:
This is the main blocker for me picking up v1 in some of the most expensive feedstocks on conda-forge, so I'm super excited about this one!

wolfv commented:
/relock

h-vetinari commented on conda_forge_tick/migrators/version.py:
```diff
@@ -197,21 +222,27 @@ def migrate(
-        )
+        recipe_dir = Path(recipe_dir)
+        recipe_file = recipe_dir / "meta.yaml"
+        recipe_yaml = recipe_dir / "recipe.yaml"
```
This naming is very confusing. Both are recipes, both are files, both are in yaml. I suggest either `meta_yaml` / `recipe_yaml` or something like `recipe_file_v0` / `recipe_file_v1`

h-vetinari replied:
If you need to assign something to the same variable, then use `recipe_file` for that variable, and where you need to distinguish, use `recipe_file_v0` and `recipe_file_v1`. The current variable naming is not suitable

h-vetinari replied:
It's weird - your comment wasn't there when I wrote mine, and I got the notification afterwards too. Presumably you had written the comment earlier but only submitted the review later? Nothing important, was just wondering why it looks like I ignored your message 😅

h-vetinari replied:
However, the naming here is still bad. Why should v0 get `file` and v1 get a completely different name? Please use consistent naming, like `recipe_file`, `recipe_file_v0` and `recipe_file_v1`.

Something like
```suggestion
        recipe_file = None  # will be determined based on version in use
        recipe_file_v0 = recipe_dir / "meta.yaml"
        recipe_file_v1 = recipe_dir / "recipe.yaml"
        if recipe_file_v0 .exists():
            recipe_file = recipe_file_v0
            [...]
       [...]
       recipe_file.write_text(updated_meta_yaml)
```

beckermr commented on conda_forge_tick/migrators/version.py:
```diff
@@ -93,11 +93,36 @@ def filter(
+                "raw_meta_yaml" in attrs
+                and "{% set version" not in attrs["raw_meta_yaml"]
+            ):
+                print("ugh")
```
Remove this.

beckermr commented on conda_forge_tick/migrators/version.py:
```diff
@@ -93,11 +93,36 @@ def filter(
+                print("ugh")
+                return True
+        elif schema_version == 1:
+            print("Checking ... ")
```
remove this

beckermr commented on conda_forge_tick/migrators/version.py:
```diff
@@ -93,11 +93,36 @@ def filter(
+        print("Still OK")
         conditional = super().filter(attrs)
-
+        print("Conditional: %r" % conditional)
```
remove this

beckermr commented on conda_forge_tick/update_recipe/version.py:
```diff
@@ -152,20 +153,35 @@ def _try_pypi_api(url_tmpl: str, context: MutableMapping, hash_type: str, cmeta:
         {nc for nc in orig_pypi_name_candidates if nc is not None and len(nc) > 0},
         key=lambda x: len(x),
     )
-    logger.info("PyPI name candidates: %s", orig_pypi_name_candidates)
```
please add back this statement

beckermr commented on conda_forge_tick/update_recipe/version.py:
```diff
@@ -557,20 +574,28 @@ def update_version_feedstock_dir(
 
 
-def _update_version_feedstock_dir_containerized(feedstock_dir, version, hash_type):
+def _update_version_feedstock_dir_containerized(
```
This function does not use the recipe_version keywork AFAICT. Should it?

wolfv replied:
Nope! :)

beckermr commented on conda_forge_tick/update_recipe/version.py:
```diff
@@ -625,7 +650,103 @@ def _update_version_feedstock_dir_containerized(feedstock_dir, version, hash_typ
 
 
-def update_version(raw_meta_yaml, version, hash_type="sha256"):
+def update_version_v1(
```
the prints here need to be debug logging statements

beckermr commented on conda_forge_tick/url_transforms.py:
```diff
@@ -132,3 +132,8 @@ def gen_transformed_urls(url):
         if new_url not in yielded:
             yield new_url
             yielded.add(new_url)
+
+
+if __name__ == "__main__":
+    url = "https://github.com/xtensor-stack/xtensor/archive/{{ version }}.tar.gz"
+    print(set(gen_transformed_urls(url)))
```
Let's remove this debugging code.

beckermr commented on conda_forge_tick/update_recipe/version.py:
```diff
@@ -773,3 +897,30 @@ def update_version(raw_meta_yaml, version, hash_type="sha256"):
     else:
         logger.critical("Recipe did not change in version migration!")
         return None, errors
+
```
Let's remove this debugging code.

beckermr replied:
Why was this resolved? We need to remove this code. The little entry points like this rot inplace.

wolfv replied:
I removed it now. I only removed similar code in another file last time - a mistake!

beckermr commented on conda_forge_tick/migrators/version.py:
```diff
@@ -197,21 +222,27 @@ def migrate(
+            )
+        elif recipe_yaml.exists():
+            recipe_file = recipe_yaml
+            updated_meta_yaml, errors = update_version_v1(
```
This routine calls into jinja2 and so needs to be executed in a container. This is not happening here AFAICT.

wolfv replied:
This code is essentially equivalent to `update_version` which is called below, outside a container etc. (uses the same Jinja eval methods).

beckermr replied:
Ah I see. The migrator is run in a container so that covers this case. However, because of the way this PR is done currently, the admin commands to update recipe versions won't work for v1. That will need to be fixed if someone wants to use that.

beckermr reviewed (changes requested):
I left several comments that need addressing.

wolfv commented:
Thanks for the review @beckermr - I'll fix things up tomorrow morning. On a high-level, do things look alright to you?

beckermr commented:
Yes this looks good. Given how it works, we can switch the v0 recipes over to this algorithm as well and that will likely be more robust than what is in the bot now. 

I just noticed you missed some of old vs new jinja2 stuff in the url transforms module. The current test doesn't hit those code paths which is why nothing weird happened.

beckermr commented:
It looks like the relock command didn't work on this PR from a fork? Let me try again to be sure.

/relock-conda

minrk commented on conda_forge_tick/update_recipe/version.py:
```diff
@@ -557,20 +575,28 @@ def update_version_feedstock_dir(
+    feedstock_dir, version, hash_type
+) -> (bool, set):
+    if os.path.join(feedstock_dir, "recipe", "recipe.yaml"):
+        return update_version_v1(feedstock_dir, version, hash_type)
```
`update_version_v1` returns the updated recipe (not a bool), it doesn't write it back to recipe.yaml. That's handled below in this funciton, but because of this `return`, the write-back never happens, so running `pixi run bump` results in no changes written to the file.

wolfv replied:
fixed, thanks!

ytausch commented on conda_forge_tick/update_recipe/version.py:
```diff
@@ -152,20 +153,36 @@ def _try_pypi_api(url_tmpl: str, context: MutableMapping, hash_type: str, cmeta:
         return None, None
```
Not your fault, but can we get a docstring for `_try_pypi_api` now that we modified it? I personally don't understand the functionality from the function name.

wolfv replied:
Done

wolfv commented:
One more issue over here: https://github.com/prefix-dev/rattler-build-conda-compat/pull/61 will release asap

ytausch commented on conda_forge_tick/update_recipe/version.py:
```diff
@@ -625,7 +649,95 @@ def _update_version_feedstock_dir_containerized(feedstock_dir, version, hash_typ
+def update_version_v1(
+    feedstock_dir: str, version: str, hash_type: str
+) -> (str | None, set[str]):
+    """Update the version in a recipe.
+
+    Parameters
+    ----------
+    feedstock_dir : str
+        The feedstock directory to update.
+    version : str
+        The new version of the recipe.
+    hash_type : str
+        The kind of hash used on the source.
+    """
```
I'd appreciate an explanation of the return value in the docstring here.

wolfv replied:
Done

ytausch commented on conda_forge_tick/update_recipe/version.py:
```diff
@@ -625,7 +649,95 @@ def _update_version_feedstock_dir_containerized(feedstock_dir, version, hash_typ
+
+
+def update_version(raw_meta_yaml, version, hash_type="sha256") -> (str, set[str]):
     """Update the version in a recipe.
```
```suggestion
    """Update the version in a v0 recipe.
```

ytausch commented on conda_forge_tick/update_recipe/version.py:
```diff
@@ -773,3 +885,32 @@ def update_version(raw_meta_yaml, version, hash_type="sha256"):
+    import argparse
+
+    # set log level to debug
+    logging.basicConfig(level=logging.DEBUG)
```
let's use `setup_logging` from `conda_forge_tick.utils` here

wolfv replied:
Good idea!

ytausch commented on tests/test_migrators.py:
```diff
@@ -466,35 +470,63 @@ def run_test_migration(
+    if recipe_version == 0:
+        tmpdir_p.joinpath("meta.yaml").write_text(inp)
+        recipe_dir = str(tmpdir_p)
+    else:
+        tmpdir_p.joinpath(".ci_support").mkdir()
+        tmpdir_p.joinpath("recipe", "recipe.yaml").write_text(inp)
+        (tmpdir_p / ".ci_support" / "linux_64_.yaml").write_text(
+            "target_platform: linux-64"
+        )
 
+        recipe_dir = str(tmpdir_p / "recipe")
```
Let's be consistent here.

```suggestion
    elif recipe_version == 1:
        tmpdir_p.joinpath(".ci_support").mkdir()
        tmpdir_p.joinpath("recipe", "recipe.yaml").write_text(inp)
        (tmpdir_p / ".ci_support" / "linux_64_.yaml").write_text(
            "target_platform: linux-64"
        )

        recipe_dir = str(tmpdir_p / "recipe")
    else:
        raise ValueError(f"Unsupported recipe version: {recipe_version}")
```

wolfv replied:
Alright! Applied suggestion.

ytausch commented:
After this is merged, is there anything left for v1 updates to be processed by the bot? Asking because this only touches the `auto-tick` step.

wolfv commented on conda_forge_tick/migrators/version.py:
```diff
@@ -93,11 +93,32 @@ def filter(
+        )
+        print("Checking schema version: %d" % schema_version)
+        if schema_version == 0:
+            if (
+                "raw_meta_yaml" in attrs
+                and "{% set version" not in attrs["raw_meta_yaml"]
+            ):
+                return True
```
I think part of the point is to make sure that we can access this attribute safely :) I changed the code slightly to make this work

wolfv commented:
@ytausch I believe that version updates should work once this migrator is enabled. Another high-value one would be the arch migrator: https://github.com/regro/cf-scripts/pull/3019

Unfortunately it was blocked because of mini-migrators ...

ytausch commented on conda_forge_tick/update_recipe/version.py:
```diff
@@ -897,7 +925,10 @@ def update_version(raw_meta_yaml, version, hash_type="sha256") -> (str, set[str]
-    logging.basicConfig(level=logging.DEBUG)
+    from conda_forge_tick.utils import setup_logging
+
+    setup_logging("INFO")
```
didn't you want to set the log level to debug?

wolfv replied:
I think info is fine, too. I left the string there so people can change it easily.

ytausch replied:
ah, then you should adjust the comment above

wolfv replied:
Done!

h-vetinari commented on conda_forge_tick/update_recipe/version.py:
```diff
@@ -625,9 +676,103 @@ def _update_version_feedstock_dir_containerized(feedstock_dir, version, hash_typ
+    from rattler_build_conda_compat.recipe_sources import render_all_sources
+
+    feedstock_dir = Path(feedstock_dir)
+    recipe_path = feedstock_dir / "recipe" / "recipe.yaml"
+    recipe_yaml = load_yaml(recipe_path.read_text())
```
Yet other variables being used here
```suggestion
    recipe_path = feedstock_dir / "recipe" / "recipe.yaml"
    recipe_text = recipe_path.read_text()
    recipe_yaml = load_yaml(recipe_text)
```

h-vetinari replied:
might also want to rename `recipe_yaml` to `recipe_dict` or `recipe_info`, because after `load_yaml`, it's not actually a YAML in any reasonable sense of the word anymore.

wolfv replied:
A (ruamel) YAML is also not exactly a dict so might be better to keep it as is.

h-vetinari commented on conda_forge_tick/update_recipe/version.py:
```diff
@@ -625,9 +676,103 @@ def _update_version_feedstock_dir_containerized(feedstock_dir, version, hash_typ
+        recipe_yaml, variants, override_version=version
+    )
+
+    # load recipe text
+    recipe_path = feedstock_dir / "recipe" / "recipe.yaml"
+    recipe = recipe_path.read_text()
```
redundant with the above
```suggestion
```

h-vetinari commented on conda_forge_tick/update_recipe/version.py:
```diff
@@ -625,9 +676,103 @@ def _update_version_feedstock_dir_containerized(feedstock_dir, version, hash_typ
+    recipe = recipe_path.read_text()
+
+    # update the version with a regex replace
+    for line in recipe.splitlines():
```
suggest to rename `recipe` (very overloaded) to `recipe_text` (more explicit)

wolfv replied:
Done

h-vetinari commented on conda_forge_tick/update_recipe/version.py:
```diff
@@ -557,15 +596,27 @@ def update_version_feedstock_dir(
+) -> (bool, set):
+    feedstock_path = Path(feedstock_dir)
+
+    recipe = feedstock_path / "recipe" / "meta.yaml"
+    if recipe.exists():
```
use the same pattern here
```suggestion
    recipe_path = None
    recipe_path_v0 = feedstock_path / "recipe" / "meta.yaml"
    if recipe_path_v0.exists():
        recipe_path = recipe_path_v0
```

h-vetinari commented on conda_forge_tick/migrators/version.py:
```diff
@@ -197,21 +221,34 @@ def migrate(
-            hash_type=hash_type,
-        )
+        recipe_dir = Path(recipe_dir)
+        recipe = None
+        recipe_v0 = recipe_dir / "meta.yaml"
+        recipe_v1 = recipe_dir / "recipe.yaml"
+        if recipe_v0.exists():
```
Please use the same name as in other parts of the implementation. I think `recipe` is too ambiguous anyway, so it's preferable to match the already existing use of `recipe_path` here, rather than vice versa.
```suggestion
        recipe_path = None
        recipe_path_v0 = recipe_dir / "meta.yaml"
        recipe_path_v1 = recipe_dir / "recipe.yaml"
        if recipe_path_v0.exists():
```

h-vetinari reviewed (changes requested):
yet more `recipe_*` variable naming consistency issues.

ytausch commented:
This might be helpful if the locking bot fails: https://github.com/conda-incubator/relock-conda/issues/245#issuecomment-2497999662

beckermr commented on conda_forge_tick/migrators/version.py:
```diff
@@ -93,11 +94,34 @@ def filter(
+            {},
+            0,
+        )
+        print("Checking schema version: %d" % schema_version)
```
```suggestion
```

wolfv replied:
thx!

h-vetinari commented on conda_forge_tick/update_recipe/version.py:
```diff
@@ -625,9 +676,103 @@ def _update_version_feedstock_dir_containerized(feedstock_dir, version, hash_typ
+        recipe_yaml, variants, override_version=version
+    )
+
+    recipe_text = recipe_path.read_text()
```
Please move that line up a bit to where `recipe_yaml` is calculated from it.

wolfv replied:
feel free to just push to this PR :)

beckermr replied:
Yeah I'm happy to wait on this until all comments are addressed. As usual, the folks wanting things done have the onus to do them. I am not 100% happy with how the interactions and pushing in both directions (so me inlcuded) have happened with this PR and v0 -> v1 tasks in general. We need to find a better way to communicate and balance things.

h-vetinari replied:
> feel free to just push to this PR :)

Alright, didn't want to assume, but fine by me. 👍 

> I am not 100% happy with how the interactions and pushing in both directions (so me inlcuded) have happened with this PR

Do you want to elaborate on that? For me it's been pretty much standard review - the work in itself is highly appreciated, and yet at the same time, we can ensure through review-response iterations that newly added code is consistent and otherwise of a high quality before merging.

beckermr replied:
Sure! There has been some back and forth around skipping edge cases / excluding things just to get PRs done. I've also felt like i've been put in the position of blocking half-completed PRs. I'd rather not have to push back at each step and instead have things fully implemented up front. 

I also think I have not been the most responsive on PRs due to my frustration and wanting to avoid the interactions.

wolfv commented:
@beckermr I used `_get_new_url_tmpl_and_hash` which calls the function you mentioned so I hope that this works :)

wolfv commented:
It does indeed miss the code about the R versions though, so that will have to be added back in. Will try to tackle this tomorrow if @mgorny doesn't get to it before me

beckermr commented:
@wolfv That is exactly the problem. We need to ensure `get_transformed_urls` doesn't break on jinja2 with `${{` instead of `{{`.

beckermr commented:
The lockfile has rattler-build-conda-compat at 1.3.1 and so everything is already up to date.

beckermr commented:
pre-commit.ci autofix

beckermr commented:
pre-commit.ci autofix

wolfv commented:
@beckermr the tests have highlighted an issue with the current `extract_requirements` function. It didn't work well with `pin_subpackage` or `pin_compatible` expressions. These are turned into dictionaries when rendering, so that the result looks something like:

```yaml
- foo
- pin_subpackage:
    name: foobar
    pin:
      upper_bound: x.x
...
```

This ends up not being hashable, and then the extraction failed somewhere. 
For now, I fixed it by turning this dictionary into a normal looking spec of `pin_subpackage {name}`

Do you happen to know what the best thing to turn this into would be? I could also just make it a requirement on the name that comes out of this.

I also added the `split` filter to the jinja env unconditionally to help the v1 recipes. I hope that's OK.

beckermr commented:
Those need to be dictionaries for other parts of the code to work. So whatever requires them to be hashable needs to be changed. We can't merge this pr as it stands now.

wolfv commented:
> We can't merge this pr as it stands now.

I didn't ask for this, I just asked for guidance :)

beckermr commented:
Ah sorry. I meant to indicate this is blocking. I suggest if rattler is that strict about its inputs that the v1 version update code should make a copy of the data it sends in, adjust it, and then undo the changes on the way out.

wolfv commented:
Sorry, I think there is a misunderstanding. What I am fixing here is code that was already present for a long time in the graph generation. This is not related to the functionality of this PR, nor rattler-build. 

I think we didn't properly test recipes with `pin_subpackage` before, and now this started to highlight a problem in the graph generation. The failure came about in this line:

https://github.com/regro/cf-scripts/pull/3525/files#diff-80dce555ca3e04ee9e79590c4b418bad646f6ff177d8332104d68c78ce75ff3cR195

Does the code really expect this to be a dictionary in other places? Ie. 

```
run:
  - pin_subpackage:
      name: foo
      pin: { bla ..}
```

As I don't think conda-build would create specs in this way.

beckermr commented:
Ah thanks for asking a good question. The code parses the name only usually (https://github.com/wolfv/cf-scripts/blob/add-version-bump-v1/conda_forge_tick/utils.py#L58) but in some cases will output a dictionary (https://github.com/wolfv/cf-scripts/blob/add-version-bump-v1/conda_forge_tick/utils.py#L78) when dealing with pinnings. I was under the impression we were working with the pinnings parsing and we wanted to change it. However, we should not be using that at all, so I think the bug is somewhere else.

beckermr commented on conda_forge_tick/feedstock_parser.py:
I don't the changes related to flattening dicts in this file are actually needed. I think there is a bug somewhere that is causing the parsing of v1 recipes to output the special pinning parsing options when it should not. In this function here: https://github.com/regro/cf-scripts/blob/main/conda_forge_tick/utils.py#L612 the dicts should be output only if `for_pinning` is set to true.

wolfv commented:
If I see correctly, then what ends up in the graph is just a simple package name: https://github.com/regro/cf-graph-countyfair/blob/8c12b88aa984748c8927302d87b4f898e32f72dd/node_attrs/6/b/e/9/0/zlib.json#L142-L145

(I believe the info comes from run_exports / pin_subpackage).

beckermr commented:
Yes. So there is a bug in the v1 parsing that is adding dicts in a case it should not. Sometimes we do output dicts, but only when for_pinning=True.

ytausch commented:
Hi, I am a bit out of the loop about what happened in this PR but please let me shamelessly advertise that #3417 adds a way to test version updates using the entire's bot pipeline, including actual PRs made to GitHub.

It's likely easy to secure this PR against subtle bugs if #3417 gets merged first.

wolfv commented:
Cool! I think the test migration that is ran in the tests of this PR are also quite good but the deeper the tests the better! 

Does your test also check if new versions are "found"? The tests in this PR only test the recipe editing essentially.

ytausch commented:
Yes, they start from an empty cf-graph and execute all the CLI entrypoints in sequence.
The search for new updates is mainly done by `update-upstream-versions` which is part of that :)

beckermr commented:
We'll want both. The integration tests handle items that happen when the bot runs live (git checkouts, etc.). However, it is faster to debug certain things with a more targeted test like is present in this PR. There is no need to couple the two.

wolfv commented:
I just released `rattler-build-conda-compat` 1.3.2 which should fix the remaining errors. It will take a little more time to propagate through the CDNs etc. But I think then this should be ready :)

wolfv commented:
Nice, tests passed 🙏

beckermr commented on conda_forge_tick/update_recipe/version.py:
```diff
@@ -134,14 +135,40 @@ def _try_url_and_hash_it(url: str, hash_type: str):
-        .from_string(tmpl)
-        .render(**context)
-    )
+    env = jinja2.Environment(undefined=jinja2.StrictUndefined)
```
```suggestion
    env = jinja2.sandbox.SandboxedEnvironment(undefined=jinja2.StrictUndefined)
```

This needs to be sandboxed.

wolfv commented:
Wow, thanks @beckermr! 

Do you know if anything else needs to happen?

beckermr commented:
Not sure. We'll find out if we get some version bumps!

wolfv commented:
I asked the bot for one here: https://github.com/conda-forge/conda-forge-webservices/actions/runs/12795837076/job/35674099360 

Do we need to update a docker image somewhere?

beckermr commented:
so to make that go live, you'll need to

1. bump conda-forge-tick-feedstock to the latest version (make sure to adjust the constraints in the recipe on rattler build conda compat when doing that)
2. update the webservices conda env

If you do 1, 2 will happen automatically.

wolfv commented:
OK, I bumped the conda-forge-tick-feedstock: https://github.com/conda-forge/conda-forge-tick-feedstock/pull/60

ytausch commented:
appears to work: https://github.com/conda-forge/subfinder-feedstock/pull/1/files

minrk commented:
absolutely wonderful, thanks folks!
